universe/node/api_space_attributes.go

```diff
@@ -68,12 +72,182 @@ func (n *Node) apiGetSpaceAttributesValue(c *gin.Context) {
+		api.AbortRequest(c, http.StatusNotFound, "space_not_found", err, n.log)
+		return
+	}
+
+	spaces := rootSpace.GetSpaces(true)
+	spaceAttributes := make(dto.SpaceAttributeValues, len(spaces))
+
+	spaces[rootSpace.GetID()] = rootSpace
+
+	for _, space := range spaces {
+		attributeID := entry.NewAttributeID(pluginID, inQuery.AttributeName)
+		attributeValue, ok := space.GetSpaceAttributeValue(attributeID)
+		if !ok {
+			err := errors.Errorf("Node: apiGetSpaceWithChildrenAttributeValues: space attribute value not found: %s", attributeID)
+			api.AbortRequest(c, http.StatusNotFound, "attribute_not_found", err, n.log)
+			return
+		}
+
+		spaceAttributes[space.GetID()] = attributeValue
+	}
+
+	c.JSON(http.StatusOK, spaceAttributes)
+}
+
+// @Summary Updates entire space attribute value
+// @Schemes
+// @Description Updates entire space attribute value
+// @Tags spaces
+// @Accept json
+// @Produce json
+// @Param space_id path string true "Space ID"
+// @Param plugin_id body string true "Plugin ID"
+// @Param attribute_name body string true "Attribute Name"
+// @Param attribute_value body []string true "[Attribute Value]"
+// @Success 202 {object} entry.AttributeValue
+// @Success 500 {object} api.HTTPError
+// @Success 400 {object} api.HTTPError
+// @Success 404 {object} api.HTTPError
+// @Router /api/v4/spaces/{space_id}/attributes [post]
+func (n *Node) apiSetSpaceAttributesValue(c *gin.Context) {
+	type Body struct {
+		PluginID       string         `json:"plugin_id" binding:"required"`
+		AttributeName  string         `json:"attribute_name" binding:"required"`
+		AttributeValue map[string]any `json:"attribute_value" binding:"required"`
+	}
+
+	inBody := Body{}
+
+	if err := c.ShouldBindJSON(&inBody); err != nil {
+		err = errors.WithMessage(err, "Node: apiSetSpaceAttributesValue: failed to bind json")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_body", err, n.log)
+		return
+	}
+
+	spaceID, err := uuid.Parse(c.Param("spaceID"))
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiSetSpaceAttributesValue: failed to parse space id")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_space_id", err, n.log)
+		return
+	}
+
+	pluginID, err := uuid.Parse(inBody.PluginID)
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiSetSpaceAttributesValue: failed to parse plugin id")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_plugin_id", err, n.log)
+		return
+	}
+
+	space, ok := n.GetSpaceFromAllSpaces(spaceID)
+	if !ok {
+		err := errors.Errorf("Node: apiSetSpaceAttributesValue: space not found: %s", spaceID)
+		api.AbortRequest(c, http.StatusNotFound, "space_not_found", err, n.log)
+		return
+	}
+
+	attributeID := entry.NewAttributeID(pluginID, inBody.AttributeName)
+
+	modifyFn := func(current *entry.AttributePayload) (*entry.AttributePayload, error) {
+		newValue := func() *entry.AttributeValue {
+			value := entry.NewAttributeValue()
+			*value = inBody.AttributeValue
+			return value
+		}
```

Comment:
you can do it without creating new value, just use inBody.AttributeValue as it is